### PR TITLE
Added filters for events listing

### PR DESCRIPTION
**Pull request by ZAINABFATIMA0**
Created following filters in events views
  - Name
  - Category
  - Registration start date
  - Registration end date
  - Event start date
  - Event end date
  - Location

### CODE REVIEW

**Review by Hamza-Chaudhary65 — APPROVED** [review]

File: events/filters.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    class Meta:
+        model = Event
+        fields = [
+            'name', 'category', 'registration_start_date', 
+            'registration_end_date', 'event_start_date', 'event_end_date', 'location'
+        ]
+
+    def filter_by_location(self, queryset, name, value):
+        try:
```
Comment (sheralim012) [inline events/filters.py:44]: try except block should be as small as possible and should only have necessary code which could raise potential error

File: events/filters.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            latitude, longitude = map(float, value.split(','))
+            user_location = Point(latitude, longitude, srid=4326)
+            distance = 5000
```
Comment (sheralim012) [inline events/filters.py]: read about django constance https://django-constance.readthedocs.io/en/latest/ and integrate that for distance settings

**Review by sheralim012 — CHANGES_REQUESTED** [review]

**Review by Hamza-Chaudhary65 — APPROVED** [review]

File: eventify/settings.py [hunk 3]
```diff
@@ -38,14 +38,30 @@
...
+CONSTANCE_CONFIG = {
+   'MAX_DISTANCE': (5000, 'Maximum distance in meters for filtering events.', int),
+}
+
+CONSTANCE_IGNORE_ADMIN_VERSION_CHECK = True
+
+CONSTANCE_REDIS_CONNECTION = {
+    'host': 'localhost',
+    'port': 6379,
+    'db': 0,
```
Comment (sheralim012) [inline eventify/settings.py]: shouldn't these be env. variables?

Comment (sheralim012) [reply eventify/settings.py]: also add readme for setting up redis

**Review by sheralim012 — CHANGES_REQUESTED** [review]

**Review by Hamza-Chaudhary65 — APPROVED** [review]

**Review by sheralim012 — APPROVED** [review]

**Pull request merged**